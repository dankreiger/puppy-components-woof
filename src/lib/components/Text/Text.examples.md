Text examples:

###### default

```js
import { Text } from 'puppy-components-woof';

<Text text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr" />;
```

###### font-size: 28px; font-weight: 300; color: #00ABE7

```js
import { Text } from 'puppy-components-woof';

<Text text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr" fontSize="28px" fontWeight="300px" color="#00ABE7" />;
```

###### with hover styles

```js
import { Text } from 'puppy-components-woof';

<Text text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr" color="#00ABE7" fontSize="28px" hoverStyles={{color: 'inherit' }} />;
```

