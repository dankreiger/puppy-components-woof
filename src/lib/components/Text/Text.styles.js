import styled, { createGlobalStyle } from 'styled-components';

export const TextGlobalStyles = createGlobalStyle`
/* reset tags */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p {
  font-size: 1em;
  font-weight: normal;
  margin: 0;
  transition: all 250ms;
  }
`;
