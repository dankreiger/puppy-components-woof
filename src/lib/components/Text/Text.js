import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { TextGlobalStyles } from './Text.styles';

const Text = ({
  color,
  className,
  fontFamily,
  fontSize,
  fontWeight,
  tag,
  text,
  customStyles,
  hoverStyles
}) => {
  /* html tag defined dynamically */
  const Tag = `${tag}`;
  const textStyles = {
    color,
    fontSize,
    fontFamily,
    fontWeight,
    ...customStyles
  };
  const hoverTextStyles = { ...textStyles, ...hoverStyles };
  const [hovered, setHovered] = useState(false);
  return (
    <>
      <TextGlobalStyles textStyles={textStyles} hoverStyles={hoverStyles} />
      <Tag
        className={className}
        style={hovered ? hoverTextStyles : textStyles}
        onMouseOver={() => setHovered(true)}
        onMouseOut={() => setHovered(false)}
      >
        {text}
      </Tag>
    </>
  );
};

Text.propTypes = {
  /** Custom class */
  className: PropTypes.string,
  /** Custom color for Text */
  color: PropTypes.string,
  /** Font family */
  fontFamily: PropTypes.string,
  /** Font weight */
  fontWeight: PropTypes.string,
  /** Font size */
  fontSize: PropTypes.string,
  /** HTML tag for Text SEO */
  tag: PropTypes.string,
  /** Text content for Text */
  text: PropTypes.string,
  /** More custom styles */
  customStyles: PropTypes.object,
  /** Hover styles */
  hoverStyles: PropTypes.object
};

Text.defaultProps = {
  tag: 'p',
  color: 'inherit',
  fontFamily: '"Prometo", Verdana, Arial, Helvetica, sans-serif',
  fontSize: '16px',
  fontWeight: '500'
};

export default Text;
