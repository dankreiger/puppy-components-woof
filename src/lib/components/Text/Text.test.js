import React from 'react';
import { shallow } from 'enzyme';

import Text from './Text';

describe('<Text />', () => {
  describe('rendering', () => {
    it('should render', () => {
      const wrapper = shallow(<Text />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('custom', () => {
    it('allows custom html tag', () => {
      let props = {
        tag: 'div'
      };
      let wrapper = shallow(<Text {...props} />);
      expect(wrapper.name()).toBe('div');

      props = {
        tag: 'div'
      };
      wrapper = shallow(<Text {...props} />);
      expect(wrapper.name()).toBe('div');
    });

    it('allows custom text', () => {
      const props = {
        text: 'The quick brown fox'
      };
      const wrapper = shallow(<Text {...props} />);
      expect(wrapper.text()).toBe(props.text);
    });

    it('allows custom color', () => {
      const props = {
        color: 'purple'
      };
      const wrapper = shallow(<Text {...props} />);
      expect(wrapper.props().style.color).toBe(props.color);
    });
  });
});
