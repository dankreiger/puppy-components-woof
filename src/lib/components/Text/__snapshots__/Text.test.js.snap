// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> rendering should render 1`] = `ShallowWrapper {}`;
